### PR TITLE
Stokhos: Provide algorithm and math overloads for Sacado::MP::Vector

### PR DESCRIPTION
**Pull request by masterleinad**
@trilinos/stokhos

## Motivation
https://github.com/kokkos/kokkos/issues/5216.](https://github.com/kokkos/kokkos/pull/5170 promoted, among others, `Kokkos::Experimental::max` to `Kokkos::max`. This caused `Stokhos_SacadoMPVectorUnitTest_MPI_1` to fail since now the `Kokkos` function was selected in overload resolution for the unqualified `max()` call. Note that this was already happening for `Experimental` backends, like `HIP` before, since then `Kokkos::Experimental::max` was selected. This pull request fixes this by providing a better overload for `Vector` that doesn't require implicit conversion from `Expr`.

The last paragraph in https://github.com/trilinos/Trilinos/issues/10740#issuecomment-1180885529 might be related.

## Related Issues

* Closes https://github.com/kokkos/kokkos/issues/5216

## Testing
Compiling and running the failing unit test with the `Serial` backend.

### CODE REVIEW

**Comment by ndellingwood** [comment]
@trilinos/stokhos @etphipp

**Review by etphipp — APPROVED** [review]

**Comment by GrahamBenHarper** [comment]
Just FYI, this is being blocked on the C++17 build by the MueLu change that got past the AT before the configuration update

**Comment by ndellingwood** [comment]
Will retest after merge of #10786

**Comment by ndellingwood** [comment]
The cuda/11.4.2 failure is a build error with Zoltan2Sphynx, I don't think that can be related to the changes in this PR, though there is not much from output to parse:

snips:
```
Zoltan2Sphynx | Error building

 "/projects/sems/install/rhel7-x86_64/sems/v2/utility/cmake/3.21.1/gcc/7.3.0/mxfpluq/bin/cmake" "--build" "." "--config" "Release" "--" "-j29" "-k" "0" 

```

https://trilinos-cdash.sandia.gov/viewBuildError.php?buildid=301212

**Pull request merged**
